morf-core/src/main/java/org/alfasoftware/morf/sql/AbstractSelectStatement.java

```diff
@@ -125,6 +161,9 @@ protected void addFields(AliasedFieldBuilder... aliasedFields) {
    * @param aliasedFields The fields to add
    */
   protected void addFields(List<? extends AliasedFieldBuilder> aliasedFields) {
+    if (AliasedField.immutableDslEnabled()) {

```

Comment:
Consider pushing this repeated block into a new method on AliasedField: assertImmutableDslEnabled()

Reply:
Couldn't find it repeated anywhere, but moved it there anyway.